library/modules/Gui.cpp

```diff
@@ -1574,6 +1707,239 @@ void Gui::showAutoAnnouncement(
+
+        if (*gamemode == game_mode::ADVENTURE && !world->units.active.empty() && world->units.active[0]->counters.unconscious > 0)
+            new_report->flags.bits.unconscious = true;
+
+        if ((*gamemode == game_mode::ADVENTURE && a_flags.bits.A_DISPLAY) || (*gamemode == game_mode::DWARF && a_flags.bits.D_DISPLAY))
+        {
+            insert_into_vector(world->status.announcements, &df::report::id, new_report);
+            new_report->flags.bits.announcement = true;
+            world->status.display_timer = r.display_timer;
+            success = true;
+        }
+    }
+
+    if (*gamemode == game_mode::DWARF)
+    {
+        if (a_flags.bits.UNIT_COMBAT_REPORT)
+        {
+            if (r.unit1 != NULL)
+            {
+                if (r.flags.bits.hostile_combat)
+                    success |= addCombatReport(r.unit1, unit_report_type::Combat, new_report_index);
+                else if (r.unit1->job.current_job != NULL && r.unit1->job.current_job->job_type == job_type::Hunt)
+                    success |= addCombatReport(r.unit1, unit_report_type::Hunting, new_report_index);
+                else
+                    success |= addCombatReport(r.unit1, unit_report_type::Sparring, new_report_index);
+            }
+
+            if (r.unit2 != NULL)
+            {
+                if (r.flags.bits.hostile_combat)
+                    success |= addCombatReport(r.unit2, unit_report_type::Combat, new_report_index);
+                else if (r.unit2->job.current_job != NULL && r.unit2->job.current_job->job_type == job_type::Hunt)
+                    success |= addCombatReport(r.unit2, unit_report_type::Hunting, new_report_index);
+                else
+                    success |= addCombatReport(r.unit2, unit_report_type::Sparring, new_report_index);
+            }
+        }
+
+        if (a_flags.bits.UNIT_COMBAT_REPORT_ALL_ACTIVE)
+        {
+            FOR_ENUM_ITEMS(unit_report_type, slot)
+            {
+                if (recent_report(r.unit1, slot))
+                    success |= addCombatReport(r.unit1, slot, new_report_index);
+                if (recent_report(r.unit2, slot))
+                    success |= addCombatReport(r.unit2, slot, new_report_index);
+            }
+        }
+    }
+
+    delete_old_reports();
+
+    if (/*debug_gamelog &&*/ success)
+        Gui::writeToGamelog(message);
+    else if (success)
+        return 10;
+    else
+        return 11;
+
+    return 0;
+}
+
+int Gui::autoDFAnnouncement(df::report_init r, string message, bool log_failures)
+{   // Prints info about failed announcements to DFHack console if log_failures is true
+    int rv = autoDFAnnouncement(r, message);
+
+    if (log_failures)
+    {
+        switch (rv)
+        {
+            case 0:
+                break; // success
+            case 1:
+                Core::print("Skipped an announcement because world->show_announcements is false:\n%s\n", message.c_str());
```

Comment:
...and if you put the enum in df-structures, you could make an attribute for these descriptions, which would make them accessible directly to C++ and Lua without needing to write an extra function.

Reply:
Given https://github.com/DFHack/dfhack/pull/2113#discussion_r867597065, I would probably shy away from this unless the messages are "useful" enough for the caller to have access to them